lib/model/query/entities.js

```diff
@@ -119,6 +120,60 @@ const processSubmissionEvent = (event) => (container) =>
 ////////////////////////////////////////////////////////////////////////////////
 // GETTING ENTITIES
 
+const _get = (includeSource) => {
+  const frames = [Entity];
+  if (includeSource) {
+    frames.push(Entity.Def.into('currentVersion'), Submission, Submission.Def.into('submissionDef'), Form);
+  } else {
+    frames.push(Entity.Def.Metadata.into('currentVersion'));
+  }
+  return extender(...frames)(Actor.into('creator'), Actor.alias('current_version_actors', 'currentVersionCreator'))((fields, extend, options, deleted = false) =>
+    sql`
+  SELECT ${fields} FROM entities
+  INNER JOIN entity_defs
+      ON entities.id = entity_defs."entityId" AND entity_defs.current
+  ${extend||sql`
+    LEFT JOIN actors ON actors.id=entities."creatorId"
+    LEFT JOIN actors current_version_actors ON current_version_actors.id=entity_defs."creatorId"
+  `}
+  ${!includeSource ? sql`` : sql`
+    LEFT JOIN submission_defs ON submission_defs.id = entity_defs."submissionDefId"
+    LEFT JOIN (
+      SELECT submissions.*, submission_defs."userAgent" FROM submissions
+      JOIN submission_defs ON submissions.id = submission_defs."submissionId" AND root
+    ) submissions ON submissions.id = submission_defs."submissionId"
+    LEFT JOIN forms ON submissions."formId" = forms.id
+  `} 
+  where ${equals(options.condition)} and entities."deletedAt" is ${deleted ? sql`not` : sql``} null
+  order by entity_defs.id, entities."createdAt" desc, entities.id desc
+`);
+};
+
+const getById = (datasetId, uuid, options = QueryOptions.none) => ({ maybeOne }) =>
+  _get(true)(maybeOne, options.withCondition({ datasetId, uuid }), isTrue(options.argData.deleted))
+    .then(map((entity) => {
+      const isSourceSubmission = !!entity.aux.currentVersion.submissionDefId;
+
+      // TODO: revisit this when working on POST /entities
+      const source = new Entity.Def.Source({
+        type: isSourceSubmission ? 'submission' : 'api',
+        details: isSourceSubmission ? {
+          xmlFormId: entity.aux.form.xmlFormId,
+          instanceId: entity.aux.submission.instanceId,
+          instanceName: entity.aux.submissionDef.instanceName
+        } : null
+      });
+
```

Comment:
in case of submission deletion, we will get null in all properties of source.detail. This needs to be revisited after POST/PUT /entities